### PR TITLE
[Resource] update testcases for resource mgmt clients

### PR DESCRIPTION
**Pull request by ChenxiJiang333**

### CODE REVIEW

File: sdk/resources/azure-mgmt-resource/tests/test_mgmt_resource_private_link.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    def setup_method(self, method):
+        self.client = self.create_mgmt_client(
+            azure.mgmt.resource.privatelinks.ResourcePrivateLinkClient
+        )
+
+    @RandomNameResourceGroupPreparer()
+    @recorded_by_proxy
+    def test_private_link_list(self):
+        result = list(self.client.resource_management_private_link.list().value)
```
Comment (msyyc) [inline sdk/resources/azure-mgmt-resource/tests/test_mgmt_resource_private_link.py:22]: why call `.value` here?

Comment (ChenxiJiang333) [reply sdk/resources/azure-mgmt-resource/tests/test_mgmt_resource_private_link.py:22]: What returns by the list() method here is not an iterable object https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/resources/azure-mgmt-resource/azure/mgmt/resource/privatelinks/v2020_05_01/models/_models_py3.py#L136

Comment (msyyc) [reply sdk/resources/azure-mgmt-resource/tests/test_mgmt_resource_private_link.py:22]: Got it then I think we don't need `list (...)` here.

Comment (ChenxiJiang333) [reply sdk/resources/azure-mgmt-resource/tests/test_mgmt_resource_private_link.py]: got it, updated

**Review by msyyc — APPROVED** [review]

**Pull request merged**